CMakeLists.txt

```
@@ -41,15 +41,44 @@ FIND_PACKAGE( DD4hep REQUIRED )
 FIND_PACKAGE( ROOT REQUIRED )
 FIND_PACKAGE( MarlinTrk REQUIRED )
 
+OPTION( USE_SDHCALCONTENT "Use SDHCALContent" OFF )
+OPTION( USE_APRILCONTENT "Use APRILContent" OFF )
 
-FOREACH( pkg PandoraSDK Marlin MarlinUtil  LCContent DD4hep ROOT MarlinTrk )
+FOREACH( pkg PandoraSDK Marlin MarlinUtil LCContent DD4hep ROOT MarlinTrk )
     IF( ${pkg}_FOUND )
         INCLUDE_DIRECTORIES( SYSTEM ${${pkg}_INCLUDE_DIRS} )
         LINK_LIBRARIES( ${${pkg}_LIBRARIES} )
         ADD_DEFINITIONS ( ${${pkg}_DEFINITIONS} )
     ENDIF()
 ENDFOREACH()
 
+IF( USE_SDHCALCONTENT )
+    FIND_PACKAGE( SDHCALContent REQUIRED )
+    IF( SDHCALContent_FOUND )
+        INCLUDE_DIRECTORIES( SYSTEM ${SDHCALContent_INCLUDE_DIRS} )
+        LINK_LIBRARIES( ${SDHCALContent_LIBRARIES} )
+        ADD_DEFINITIONS ( ${SDHCALContent_DEFINITIONS} )
+        ADD_DEFINITIONS( "-DSDHCALCONTENT" )
+    ENDIF()
+ENDIF()
+
+IF( USE_APRILCONTENT )
+    SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+    FIND_PACKAGE( APRILContent REQUIRED )
+    FIND_PACKAGE( mlpack REQUIRED )
+    IF( APRILContent_FOUND )
+        INCLUDE_DIRECTORIES( SYSTEM ${APRILContent_INCLUDE_DIRS} )
+        LINK_LIBRARIES( ${APRILContent_LIBRARIES} )
+        ADD_DEFINITIONS ( ${APRILContent_DEFINITIONS} )
+        ADD_DEFINITIONS( "-DAPRILCONTENT" )
+    ENDIF()
+    IF( mlpack_FOUND )
+        INCLUDE_DIRECTORIES( SYSTEM ${mlpack_INCLUDE_DIRS} )
+        LINK_LIBRARIES( ${mlpack_LIBRARIES} )
+        ADD_DEFINITIONS ( ${mlpack_DEFINITIONS} )
+    ENDIF()
```

```suggestion
    INCLUDE_DIRECTORIES( SYSTEM ${APRILContent_INCLUDE_DIRS} )
    LINK_LIBRARIES( ${APRILContent_LIBRARIES} )
    ADD_DEFINITIONS ( ${APRILContent_DEFINITIONS} )
    ADD_DEFINITIONS( "-DAPRILCONTENT" )

    INCLUDE_DIRECTORIES( SYSTEM ${mlpack_INCLUDE_DIRS} )
    LINK_LIBRARIES( ${mlpack_LIBRARIES} )
    ADD_DEFINITIONS ( ${mlpack_DEFINITIONS} )
```
If you have `REQUIRED` in your `find_package` call there is no need to check again if it is actually found, because CMake will actually stop if it is not found.

---

```suggestion
    IF( APRILContent_FOUND )
        INCLUDE_DIRECTORIES( SYSTEM ${APRILContent_INCLUDE_DIRS} )
        LINK_LIBRARIES( ${APRILContent_LIBRARIES} )
        ADD_DEFINITIONS ( ${APRILContent_DEFINITIONS} )
        ADD_DEFINITIONS( "APRILCONTENT" )
    ENDIF()
    IF( mlpack_FOUND )
        INCLUDE_DIRECTORIES( SYSTEM ${mlpack_INCLUDE_DIRS} )
        LINK_LIBRARIES( ${mlpack_LIBRARIES} )
        ADD_DEFINITIONS ( ${mlpack_DEFINITIONS} )
    ENDIF()
```

---

Deleted the IF(... FOUND) in the commit "CMakeLists updates" (d49c7a0)

I tried to delete the `-D` in the add definitions but it seems that the compiler wasn't understanding the definition as a macro anymore and was trying to find it as file or directory, thus creating errors during compilation

---

The special handling of adding `-D` might only be available for `target_compile_definitions`(?) @andresailer